File: GO/MostFrequentDays_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import (
	"reflect"
	"testing"
)

func TestMostFrequentDays(t *testing.T) {
	tables := []struct {
		year int
		want []string
	}{
		{2427, []string{"Friday"}},
		{2185, []string{"Saturday"}},
		{1084, []string{"Tuesday", "Wednesday"}},
		{1167, []string{"Sunday"}},
		{1216, []string{"Friday", "Saturday"}},
		{1492, []string{"Friday", "Saturday"}},
		{1770, []string{"Monday"}},
		{1785, []string{"Saturday"}},
		{212, []string{"Wednesday", "Thursday"}},
	}

	for _, table := range tables {
		got := mostFrequentDays(table.year)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %v but the return was this: %v", table.want, got)
		}
	}
}

File: PHP/MostFrequentDaysTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("MostFrequentDays.php");

class MostFrequentDaysTest extends TestCase {

    public function testMostFrequentDays() {
        $obj = new MostFrequentDays();
        //Test 1
        $want = ["Friday"];
        $got = $obj->mostFrequentDays( 2427 );
        $this->assertEquals($want, $got);
        //Test 2
        $want = ["Saturday"];
        $got = $obj->mostFrequentDays( 2185 );
        $this->assertEquals($want, $got);
        //Test 3
        $want = ["Tuesday", "Wednesday"];
        $got = $obj->mostFrequentDays( 1084 );
        $this->assertEquals($want, $got);
        //Test 4
        $want = ["Sunday"];
        $got = $obj->mostFrequentDays( 1167 );
        $this->assertEquals($want, $got);
        //Test 5
        $want = ["Friday", "Saturday"];
        $got = $obj->mostFrequentDays( 1216 );
        $this->assertEquals($want, $got);
        //Test 6
        $want = ["Friday", "Saturday"];
        $got = $obj->mostFrequentDays( 1492 );
        $this->assertEquals($want, $got);
        //Test 7
        $want = ["Monday"];
        $got = $obj->mostFrequentDays( 1770 );
        $this->assertEquals($want, $got);
        //Test 8
        $want = ["Saturday"];
        $got = $obj->mostFrequentDays( 1785 );
        $this->assertEquals($want, $got);
        //Test 9
        $want = ["Wednesday", "Thursday"];
        $got = $obj->mostFrequentDays( 212 );
        $this->assertEquals($want, $got);
    }

}
